tutorials/kube-mqtt/index.md

```diff
@@ -0,0 +1,267 @@
+
+	cd refresher-container
+	gcloud builds submit --tag gcr.io/$PROJECT/refresher .
+
+## Create Cloud resources:
+
+### Create the device keys
+
+	cd ../bridge
+	openssl ecparam -genkey -name prime256v1 -noout -out ec_private.pem
+	openssl ec -in ec_private.pem -pubout -out ec_public.pem
+
+### Create IoT Core resources
+
+	gcloud pubsub topics create device-events
+	gcloud pubsub subscriptions create debug --topic device-events
+	
+	gcloud iot registries create $REGISTRY \
+	--region=$REGION \
+	--event-notification-config topic=device-events
+	
+	gcloud iot devices create $DEVICE \
+	--region $REGION \
+	--registry $REGISTRY \
+	--public-key path=ec_public.pem,type=es256-pem
+
+This creates an IoT Core device called "bridge" that you will use to authenticate the bridge to IoT Core with.
+
+The pubsub topics are used to verify that data is being bridged to cloud correctly.
+
+## Deploy the bridge to Kubernetes
+
+### Update project specific settings
+
+Get the specific address for the bridge manager container:
+
+	gcloud container images describe gcr.io/$PROJECT/refresher --format="value(image_summary.fully_qualified_digest)"
+
+In the repo file `bridge/project-image.yaml`, add the output from the previous command as `image` .
+
+If running the Kubernetes cluster outside the Cloud Project, make the images public:
+
+	gsutil iam ch allUsers:objectViewer gs://artifacts.$PROJECT.appspot.com
+
+Note this will make the container registry for the project public.
+
+In production environments, you would provision the cluster with the required [image pull secrets](http://docs.heptio.com/content/private-registries/pr-gcr.html).
+
+Edit PROJECT_ID value in `device-config.yaml`
+
+### Deploy
+
+You are going to use the built in version of [kustomize](https://kustomize.io/) in recent kubectl versions to deploy the solution to kubernetes.
+
+Make sure you are in the base directory for this tutorial: `community/tutorials/kube-mqtt`. More about the deployment is explained in the following section, for now, deploy the workload with:
+	
+	kubectl apply -k bridge/
+
+## Understanding the bridge deployment
+
+Lets explain a few parts of this deployment
+	
+### Use of kustomize
+
+The layout of this project uses a base folder which contains default resources including yaml files for a deployment with 2 containers and a service, as well as a kustomize generated configmap.
+
+The bridge folder in the repo acts as an overlay which:
+
+ - merges the device private key file into a configmap defined in the base
+ - patches a container image with a project specific address
+ - adds project specific environment variables
+
+By keeping these more variable parts in an overlay - the based might be managed separates, or may be shared among several different overlay variants. For example, when following this tutorial you only need to change the project-id in the device-config.yaml file, but you could use different registries or device-id values if you changed those parts.
```

Comment:
This sentence is a little confusing to me:

I'm not sure I am familiar with the terms: 'based' and 'managed separates'

By keeping these variable parts in an overlay, the bridge might be managed separately or may be shared among different variants. In this way, changing the project-id in a single configuration could affect all variants while still being able to preserve different registry or device values in parts changed locally.